include/tesseract/baseapi.h

```diff
@@ -766,7 +768,9 @@ class TESS_API TessBaseAPI {
 protected:
   Tesseract *tesseract_;          ///< The underlying data object.
   Tesseract *osd_tesseract_;      ///< For orientation & script detection.
+#ifndef DISABLED_LEGACY_ENGINE
```

Comment:
`osd_tesseract_` could be disabled, too.

Reply:
Well, you can't just disable it here, because it is needed in other places, so more ifdefs are needed.

I did disable **some** more osd parts because after I managed the `equ_detect_ ` delete issue, clang will probably have the same issue with osd. Note that with the disable legacy mode, `osd_tesseract_` is always `nullptr` so there is noting to delete.

Reply:
BTW, I tested this PR with clang++ on Ubuntu 20.04 with `--diable-legacy` and the compilation completed with no crash or errors.